### PR TITLE
Fix Eclipse specific compilation issue

### PR DESCRIPTION
**Pull request by cbuescher**
Without pulling out the supplier function to the enclosing class, Eclipse 4.8
complains with the following error "No enclosing instance of type
CoordinatorTests.Cluster is available due to some intermediate constructor
invocation"

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java [hunk 1]
```diff
@@ -1101,6 +1102,9 @@ private static int defaultInt(Setting<Integer> setting) {
         private final Set<String> blackholedNodes = new HashSet<>();
         private final Map<Long, ClusterState> committedStatesByVersion = new HashMap<>();
 
+        private final Function<DiscoveryNode, PersistedState> persistedStateSupplier = localNode -> new MockPersistedState(0L,
```
Comment (ywelsch) [inline server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java]: can you call this `defaultPersistedStateSupplier`?

**Review by ywelsch — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java (hunk 1 above)
Comment (cbuescher) [reply server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java]: sure

**Comment by cbuescher** [comment]
@elasticmachine run the gradle build tests 1

**Comment by cbuescher** [comment]
@elasticmachine run default distro tests

**Pull request merged**